src/modules/firewall/src/lib/Firewall.cpp

```diff
@@ -407,6 +402,11 @@ GenericRule& GenericRule::Parse(const rapidjson::Value& value)
         m_parseError.push_back("Rule JSON is not an object");
     }
 
+    for (auto& error : m_parseError)
+    {
+        OsConfigLogError(FirewallLog::Get(), "%s", error.c_str());
```

Comment:
> OsConfigLogError(FirewallLog::Get(), "%s", error.c_str()); [](http://example.com/codeflow?start=7&length=59)

Can this log during MpiGet, MpiSetDesired or MpiGetDesired? If so, protect if 'if full logging is enabled'.

Same for all other instances.